lib/msal-node/src/network/HttpClient.ts

```diff
@@ -175,15 +176,12 @@ const networkRequestViaProxy = <T>(
                     body: JSON.parse(body) as T,
                     status: statusCode as number,
                 };
-
                 if ((statusCode < 200 || statusCode > 299) &&
```

Comment:
I thought we agreed to delete this entire if check?

Reply:
@Robbie-Microsoft I believe that only not rejecting while keeping this check intact produces the desired behavior, are there any specific conditions you foresee where this would result in premature termination/ masking of underlying error?